### PR TITLE
Fixed custom mode + 2 small additions

### PR DESCRIPTION
**Pull request by Black09**
The custom mode wasn't working properly because there was still a check for 
_self.medianame_ instead of _self.dbid_.

I also added an option to show the image selection directly when using gui mode. E.g.

_RunScript(script.artwork.downloader,mode=gui,mediatype=movie,dbid=$INFO[ListItem.DBID],poster)_

will directly open the image selection for type 'poster'.

I also added a setting which disables the ok dialog and "downloaded: 0..." notification if the user doesn't choose an image and closes the dialog.

### CODE REVIEW

**Comment by Black09** [comment]
One other thing... I commented out the AnimeWindowXMLDialogClose skin setting (line 839 in default.py) because I don't think it's needed anymore. I think it was added because of the close animation bug with script dialogs, was it? Since that's fixed, there's no need for this setting anymore. Or does it have another reason?

File: resources/settings.xml [hunk 1]
```diff
@@ -96,6 +96,7 @@
         <setting label="32036" type="bool"      id="centralize_enable"          default="false"/>
         <setting label="32038" type="folder"    id="centralfolder_movies"       default=""      subsetting="true"    source="auto" option="writeable" enable="eq(-1,true)" visible="eq(-1,true)"/>
         <setting label="32037" type="folder"    id="centralfolder_tvshows"      default=""      subsetting="true"    source="auto" option="writeable" enable="eq(-2,true)" visible="eq(-2,true)"/>
+        <setting label="32104" type="bool"      id="show_cancellation_info"     default="true"/>
```
Comment (MartijnKaijser) [inline resources/settings.xml]: this can be removed as there's no need for the option imo. just don't show the message

File: default.py [hunk 2]
```diff
@@ -854,8 +890,9 @@ def MyDialog(self, image_list):
 class MainGui(xbmcgui.WindowXMLDialog):
     def __init__(self, *args, **kwargs):
         xbmcgui.WindowXMLDialog.__init__(self)
-        xbmc.executebuiltin('Skin.Reset(AnimeWindowXMLDialogClose)')
-        xbmc.executebuiltin('Skin.SetBool(AnimeWindowXMLDialogClose)')
+        # This makes no sense
+        # xbmc.executebuiltin('Skin.Reset(AnimeWindowXMLDialogClose)')
+        # xbmc.executebuiltin('Skin.SetBool(AnimeWindowXMLDialogClose)')
```
Comment (MartijnKaijser) [inline default.py]: remove it complete

File: resources/lib/utils.py [hunk 3]
```diff
@@ -114,7 +115,8 @@ def dialog_msg(action,
                 msg = line1
             else:
                 msg = line1 + ': ' + line2
-            xbmc.executebuiltin("XBMC.Notification(%s, %s, 7500, %s)" % (line0, msg, __icon__))
+            if cancelled == False:
```
Comment (MartijnKaijser) [inline resources/lib/utils.py:118]: if not cancelled:   should suffice :)

**Comment by MartijnKaijser** [comment]
Will look through the rest tomorrow :)
thx in advance

**Comment by MartijnKaijser** [comment]
Tested it and works ok afaik. Can you squash it down to just one commit?

**Comment by Black09** [comment]
Don't know if that worked correctly... if not, it's maybe easier to re-fork, add the changes and make a new pull request.

**Comment by MartijnKaijser** [comment]
I've pulled it in manually :)
thx

**Comment by Black09** [comment]
ok :)

**Pull request closed without merging**